vms/platformvm/vm.go

```diff
@@ -274,6 +280,46 @@ func (vm *VM) Initialize(
 	return nil
 }
 
+func (vm *VM) startMempoolPruner(frequency time.Duration) {
```

Comment:
I feel like `start` makes it sound like this function will create the goroutine and exit... should we call this `periodicallyPruneMempool` or something?